app/models/teacher.rb

```diff
@@ -0,0 +1,8 @@
+class Teacher < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :recoverable, :rememberable, :timeoutable, :trackable and :omniauthable
+
+  devise :database_authenticatable, :registerable, :validatable
+
+  validates_presence_of :first_name, :last_name, :password_confirmation
```

Comment:
I understand the `first_name` and `last_name`, but why `password_confirmation`? 🤔

Reply:
Except if I remove password_confirmation, no value will be passed from FaktoryBot to the models/teacher_spec.rb (though don't know why). So i leave it as it is.

Reply:
I mean, Marta, it only makes sense :D If you remove `validates :password_confirmation, presence: true` from a model, then also you should remove the test testing it: `it { is_expected.to validate_presence_of(:password_confirmation) }. This logic is just not needed and could later shoot you in the foot :)

Reply:
Wrong thread, but I couldn't find reply button to Your last comment on .gitignore. You mentioned there that password_confirmation really needs to go, but I removed it already from model, factory and model test. Do you mean I have to remove test for password too? Sorry for the trouble, but apparently my brain is in slumber.

Reply:
oh damn, sorry Marta, I didn't realise this file was updated! I checked just now, all good :) I'm resolving this thread, thank you!